### PR TITLE
fix(elasticsearch): set tags on span, regardless of sampling

### PR DESCRIPTION
**Pull request by tabgok**
The Datadog ElasticSearch integration avoids setting metadata on spans when those spans are marked to be dropped.  A recent change was added for sampling to lazily determine whether a span is sampled - which means the sample decision is much too late for this code (see https://github.com/DataDog/dd-trace-py/pull/8308).

Prior to this fix, some elasticsearch query spans incorrectly failed to set metadata based upon sampling rules, and thus incorrectly participated in metrics.

After this fix, the metadata is set regardless of sampling decisions.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: releasenotes/notes/fix-elasticsearch-always-populate-span-tags-c2a212ec706b3f4b.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    elasticsearch: this fix resolves an issue where span tags were not fully populated on "sampled" spans, causing metric dimensions to be incorrect when spans were prematurely marked as sampled.
```
Comment (wantsui) [inline releasenotes/notes/fix-elasticsearch-always-populate-span-tags-c2a212ec706b3f4b.yaml:4]: Do the metric dimensions here refer to resource names?

**Review by wantsui — APPROVED** [review]
LGTM (just a tiny note about the release notes for clarity)!

File: releasenotes/notes/fix-elasticsearch-always-populate-span-tags-c2a212ec706b3f4b.yaml (hunk 1 above)
Comment (tabgok) [reply releasenotes/notes/fix-elasticsearch-always-populate-span-tags-c2a212ec706b3f4b.yaml:4]: This particular customer cares about resource names, but generally there are several fields which were not calculated

Comment (wantsui) [reply releasenotes/notes/fix-elasticsearch-always-populate-span-tags-c2a212ec706b3f4b.yaml]: Makes sense, the new edit looks good to me too.

File: ddtrace/contrib/internal/elasticsearch/patch.py [hunk 2]
```diff
@@ -142,11 +142,6 @@ def _perform_request(func, instance, args, kwargs):
 
             span.set_tag(SPAN_MEASURED_KEY)
 
-            # Only instrument if trace is sampled or if we haven't tried to sample yet
-            if span.context.sampling_priority is not None and span.context.sampling_priority <= 0:
```
Comment (mabdinur) [inline ddtrace/contrib/internal/elasticsearch/patch.py:146]: We also have this logic in the new avro integration: https://github.com/DataDog/dd-trace-py/blob/v2.14.2/ddtrace/contrib/internal/avro/schema_iterator.py#L96 and protobuf integrations: https://github.com/DataDog/dd-trace-py/blob/v2.14.2/ddtrace/contrib/internal/protobuf/schema_iterator.py#L76. 

The algosearch: https://github.com/DataDog/dd-trace-py/blob/v2.14.2/ddtrace/contrib/internal/algoliasearch/patch.py#L133 and postgres integrations are also impacted: https://github.com/DataDog/dd-trace-py/blob/v2.14.2/ddtrace/contrib/internal/psycopg/extensions.py#L46

**Review by mabdinur — APPROVED** [review]
This change looks good. We may also need to apply this fix to other integrations

File: ddtrace/contrib/internal/elasticsearch/patch.py (hunk 2 above)
Comment (tabgok) [reply ddtrace/contrib/internal/elasticsearch/patch.py:146]: Thanks! I'll create a follow-up PR for those other integrations

**Pull request merged**